src/test/java/uk/gov/hmcts/reform/sendletter/tasks/SerialTaskRunnerLockingTest.java

```diff
@@ -0,0 +1,153 @@
+package uk.gov.hmcts.reform.sendletter.tasks;
+
+import org.junit.Before;
+import org.junit.Test;
+import org.junit.runner.RunWith;
+import org.mockito.Mock;
+import org.mockito.runners.MockitoJUnitRunner;
+
+import java.sql.Connection;
+import java.sql.ResultSet;
+import java.sql.SQLException;
+import java.sql.Statement;
+import javax.sql.DataSource;
+
+import static org.mockito.Matchers.startsWith;
+import static org.mockito.Mockito.mock;
+import static org.mockito.Mockito.when;
+
+@RunWith(MockitoJUnitRunner.class)
+public class SerialTaskRunnerLockingTest {
+
+    @Mock
+    private DataSource source;
+    @Mock
+    private Connection connection;
+    @Mock
+    private Statement statement;
+    private SerialTaskRunner taskRunner;
+
+    @Before
+    public void setup() throws SQLException {
+        when(connection.createStatement()).thenReturn(statement);
+        when(source.getConnection()).thenReturn(connection);
+
+        taskRunner = SerialTaskRunner.get(source);
+    }
+
+    @Test
+    public void successfully_locks_and_unlocks() throws SQLException {
+        lockedWithSuccess();
+        unlockedWithSuccess();
+
+        taskRunner.tryRun(1, () -> {
+        });
+    }
+
+    @Test
+    public void successfully_locks_and_fails_to_unlock() throws SQLException {
+        lockedWithSuccess();
+        unlockedWithFailure();
+
+        taskRunner.tryRun(1, () -> {
+        });
+    }
+
+    @Test
+    public void fails_to_lock_and_unlock() throws SQLException {
```

Comment:
can we name the test to describe the behaviour of tested object instead of mocks setup?

Reply:
for example `should_x_when_y`

Reply:
Exactly - should_{expected_behaviour}_when_{circumstances}

Reply:
Thanks for all the comments. I have renamed the test methods so that they could be used as 
fact statements in conjunction with the test object, i.e.:

object _throws_exception_when_given_invalid_argument()_